metrics-operator/controllers/common/providers/dynatrace/common.go

```diff
@@ -24,20 +25,23 @@ type Error struct {
 	Message string `json:"message"`
 }
 
-func getDTSecret(ctx context.Context, provider metricsapi.KeptnMetricsProvider, k8sClient client.Client) (string, error) {
+func getDTSecret(ctx context.Context, provider metricsapi.KeptnMetricsProvider, k8sClient client.Client) (dynatraceclient.SecretValues, error) {
```

Comment:
Since the `dynatrace` provider and `dql` provider will require different secret formats, I suggest to create a new `getDQLSecret()` function where the new structure is being used, and make the `dql` provider use that one to retrieve the secret.
The `dynatrace` provider should keep using the `getDTSecret()` function which can be left unchanged